onmt/Models.py

```diff
@@ -425,6 +429,117 @@ def forward(self, src, tgt, lengths, dec_state=None):
             attns = None
         return out, attns, dec_state
 
+    @property
```

Comment:
No I disagree with this. None of this should be in Models.py optimization should be basically independent of models.